### PR TITLE
🌱 Bump code-generator to v0.23.0

### PR DESCRIPTION
**Pull request by fmuyassarov**
**What this PR does / why we need it**:
Fixes the issues mentioned in https://github.com/metal3-io/cluster-api-provider-metal3/pull/357#issuecomment-988278632.

### CODE REVIEW

File: hack/tools/go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/golang/mock v1.6.0
 	github.com/golangci/golangci-lint v1.43.0
 	github.com/jteeuwen/go-bindata v3.0.7+incompatible
-	k8s.io/code-generator v0.22.2
+	k8s.io/code-generator v0.23.0
```
Comment (furkatgofurov7) [inline hack/tools/go.mod:10]: should not this be `v0.23.0-alpha.4` as per linked comment in PR description?

**Review by furkatgofurov7 — COMMENTED** [review]
Also there is an empty `test.txt` file in the commit which most probably added accidentally

File: hack/tools/go.mod (hunk 1 above)
Comment (fmuyassarov) [reply hack/tools/go.mod:10]: The release of 1.23.0 is out now, so I think that's okay?

**Review by kashifest — APPROVED** [review]

File: hack/tools/go.mod (hunk 1 above)
Comment (furkatgofurov7) [reply hack/tools/go.mod:10]: Okay

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Pull request merged**